### PR TITLE
Cappl 391 wire in workflow registry

### PR DESCRIPTION
**Pull request by ettec**
Resolves-> https://smartcontract-it.atlassian.net/browse/CAPPL-139

### CODE REVIEW

File: core/services/chainlink/application.go [hunk 1]
```diff
@@ -256,32 +272,78 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
...
+				}
+
+				outgoingConnectorHandler, err := webapi.NewOutgoingConnectorHandler(connector,
+					webAPIConfig,
+					capabilities2.MethodWebAPITarget, webAPILggr)
+				if err != nil {
+					return nil, fmt.Errorf("could not create outgoing connector handler: %w", err)
+				}
+
+				eventHandler := syncer.NewEventHandler(globalLogger, syncer.NewWorkflowRegistryDS(opts.DS, globalLogger), syncer.NewFetcherFunc(globalLogger, outgoingConnectorHandler), workflowstore.NewDBStore(opts.DS, globalLogger, clockwork.NewFakeClock()), opts.CapabilitiesRegistry,
```
Comment (cedric-cordenier) [inline core/services/chainlink/application.go]: NewFakeClock() -> NewRealClock() :)

File: core/services/chainlink/application.go [hunk 2]
```diff
@@ -256,32 +272,78 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
...
+					custmsg.NewLabeler(), clockwork.NewRealClock(), keys[0])
+
+				loader := syncer.NewWorkflowRegistryContractLoader(cfg.Capabilities().WorkflowRegistry().Address(), func(ctx context.Context, bytes []byte) (syncer.ContractReader, error) {
+					return relayer.NewContractReader(ctx, bytes)
```
Comment (cedric-cordenier) [inline core/services/chainlink/application.go:330]: You can simplify this to just relayer.NewContractReader I think?

File: core/services/chainlink/application.go [hunk 3]
```diff
@@ -256,32 +272,78 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
...
+				}, eventHandler)
+
+				wfSyncer := syncer.NewWorkflowRegistry(globalLogger, func(ctx context.Context, bytes []byte) (syncer.ContractReader, error) {
+					return relayer.NewContractReader(ctx, bytes)
```
Comment (cedric-cordenier) [inline core/services/chainlink/application.go:334]: Same with this

File: core/services/chainlink/application.go (hunk 2 above)
Comment (ettec) [reply core/services/chainlink/application.go:330]: it needs to be the factory method, internally the workflow registry create this on demand am assuming for good reason

File: core/services/chainlink/application.go (hunk 1 above)
Comment (ettec) [reply core/services/chainlink/application.go]: ah. good spot. sorted

**Review by cedric-cordenier — DISMISSED** [review]

**Review by kidambisrinivas — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]

**Review by EasterTheBunny — APPROVED** [review]

**Pull request merged**